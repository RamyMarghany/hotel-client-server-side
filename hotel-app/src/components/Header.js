import React, {Component} from 'react'
import {Link} from 'react-router-dom'
import User from '../UserUtilities'
import '../styles/main.scss'

 class  Header extends Component {
     state = {
         username:'',
         password:'',
         isLoggedIn: false
     }

     validateForm(){
        return this.state.username.length>0 && this.state.password.length > 0
    }
    handleChange = (e) =>{
        this.setState({
            [e.target.name]: e.target.value
        })
    }
    handleSubmit = (e) =>{
        e.preventDefault();
        User.logIn(this.state.username,this.state.password)
        this.setState({
            isLoggedIn:true
        })
    }

    handleLogout = (e) =>{
        User.logOut()
        this.setState({
            username:'',
            password:'',
            isLoggedIn:false
        })
    }

     render(){
        return (
            <div className="header-nav">
               <Link to={'/'} className="logo-link">
                <img 
                    src={require(`../assets/trivago-logo-1.png`)} 
                    alt="{trivago}" className="logo"
                    /> 
               </Link>
               
            {this.state.isLoggedIn ? (
                   <div className="logout-form">
                        <p className="username"><span className="welcome">Welcome,</span> {this.state.username.split('@')[0]}</p>
                        <button onClick={this.handleLogout} className="logout-btn">log Out</button>
                   </div>
                )
               :(
                   <div className="login-form">
                        <input 
                            type="text"
                            name="username"
                            value={this.state.username}
                            onChange={this.handleChange}
                            placeholder="username"
                        />
                        <input 
                            type="password"
                            name="password"
                            value={this.state.password}
                            onChange={this.handleChange}
                            placeholder="password"
                        />
                    <button onClick={this.handleSubmit} disabled={!this.validateForm()} className="login-btn">Login</button>
               </div>
               )
            } 
            </div>
          )
     }
   
}

export default Header
