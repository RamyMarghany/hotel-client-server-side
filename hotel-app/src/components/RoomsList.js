import React, { Component } from 'react'
import axios from 'axios';

const getRoomsByHotelID = (hotelId) => {
    return new Promise((resolve, reject)=>{
        axios.get(`http://localhost:3000/rooms?hotelId=${hotelId}`)
        .then((response) => {
            resolve(response.data)
         })
        .catch(reject)
      })
    }
    
class RoomsList extends Component {
    state={
        rooms:[],
        all: false
    }
    componentDidMount(){
        const {hotelId} = this.props;
        console.log(hotelId) 
        getRoomsByHotelID(hotelId)
        .then(res => {
            this.setState(()=>{
            return {rooms: res.sort((a,b)=>a.price_in_usd-b.price_in_usd)}
          })
        })
    }
    getAllRooms = () =>{
        this.setState({
            all: true
        })
    }
  render() {
      const roomsList = this.state.rooms.map( room => (
          <div className="room-container" key={room.id}>
              <h3 className="room-name">{room.name}</h3>
              <p className="room-description">{room.description}</p>
              <p className="room-price"><span>price: </span>{room.price_in_usd}$</p>
              <p className="room-capacity"><span>max occupancy: </span>{room.max_occupancy}</p>
          </div>
      ))
    return (
      <div className="rooms-list">
        <h3 className="rooms-title">hotel rooms</h3>
        {this.state.all ? roomsList : roomsList.slice(0,2) }
        <button onClick={this.getAllRooms} className="show-all-rooms">Show all rooms</button>
      </div>
    )
  }
}

export default RoomsList
