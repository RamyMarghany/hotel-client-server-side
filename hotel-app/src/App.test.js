import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import RoomsList from './components/RoomsList';
import Header from './components/Header';


// test cases for room list
describe('<RoomsList />',() => {
  const wrapper = shallow(<RoomsList/>);
  it('should be h3 with hotel rooms text',() =>{
    expect(wrapper.find('h3').text()).toBe('hotel rooms'); 
 })
 it('call componentDidMount', () => {  
  jest.spyOn(RoomsList.prototype, 'componentDidMount')
  const wrapper = shallow(<RoomsList/>)
  expect(RoomsList.prototype.componentDidMount.mock.calls.length).toBe(1)
})
it('update the check the room list come empty firstly', () =>{ 
     const wrapper = shallow(<RoomsList/>)
     expect(wrapper.find('rooms').length).toBe(0)
   })
})


 // test cases for Header
describe('<Header />',() => {
    const wrapper = shallow(<Header/>);
    it('should be .header-nav to be exist',() =>{
      expect(wrapper.find('.header-nav').exists()).toBe(true);
   })

   it('on button click to change p text', ()=>{
    const wrapper= shallow(<Header/>)
    expect(wrapper.find('.login-btn').text()).toBe('Login'); 
  })

  it('matches the snapshot',()=>{
    const tree= shallow(<Header/>);
    expect(toJson(tree)).toMatchSnapshot();
  })
})

