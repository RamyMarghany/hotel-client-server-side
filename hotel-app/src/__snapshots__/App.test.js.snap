// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches the snapshot 1`] = `
<div
  className="header-nav"
>
  <Link
    className="logo-link"
    to="/"
  >
    <img
      alt="{trivago}"
      className="logo"
      src="trivago-logo-1.png"
    />
  </Link>
  <div
    className="login-form"
  >
    <input
      name="username"
      onChange={[Function]}
      placeholder="username"
      type="text"
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <button
      className="login-btn"
      disabled={true}
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
